src/common/ConnectedHost.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { BaseStore } from './BaseStore';
+import { BaseComponent, autobind, assign, shallowCompare } from '../Utilities';
+
+// Track all components that require changes.
+let _changedComponents: ConnectedHost[];
+
+export interface IConnectedHostProps {
+  componentProps: any;
+  storesToSubscribe: string[];
+  component: any;
+  getProps: (stores: any, props: any) => any;
+}
+
+export interface IConnectedHostState {
+  props: any;
+}
+
+export class ConnectedHost extends BaseComponent<IConnectedHostProps, IConnectedHostState> {
+  public static contextTypes = {
+    stores: React.PropTypes.object
+  };
+
+  public context: {
+    stores: {
```

Comment:
We should define an `IStoreCollection` interface or similar for the mappings, and some methods to assemble and extract them. We should support the concept of a `StoreKey<T>` so you can call `getStore(itemStoreKey, this.context)` and guarantee return of the correct store type.